extensions/mongodb-client/deployment/src/test/java/io/quarkus/mongodb/NamedMongoClientConfigTest.java

```diff
@@ -0,0 +1,42 @@
+package io.quarkus.mongodb;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import javax.inject.Inject;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+
+public class NamedMongoClientConfigTest extends MongoWithReplicasTestBase {
+
+    @Inject
+    @io.quarkus.mongodb.runtime.MongoClient("cluster1")
```

Comment:
I don't think we need the entire package name, no?

Reply:
I have switched to reactive client so I can remove long path because we avoid conflict between
com.mongodb.client.MongoClient and  io.quarkus.mongodb.runtime.MongoClient.
Btw, on agroal, we have not the issue because it is AgroalDatasource vs Datasource